app/jobs/populate_fake_data_job.rb

```diff
@@ -0,0 +1,285 @@
+        api_version: 'v3',
+      },
+
+      ongoing_blood_sugar: {
+        time_fn: -> { Faker::Time.between(from: 3.month.ago, to: Time.now) },
+        size_fn: -> { rand(1..3) },
+        build_fn: -> (args) {
+          build_blood_sugar_payload(FactoryBot.build(
+            :blood_sugar,
+            :with_hba1c,
+            patient: args[:patient],
+            user: user,
+            recorded_at: args[:time_fn].call,
+            facility: user.facility
+          ))
+        },
+        patient_sample_size: 0.20,
+        request_key: :blood_sugars,
+        api_version: 'v4'
+      },
+
+      retroactive_blood_sugar: {
+        time_fn: -> { Faker::Time.between(from: 9.months.ago, to: 1.month.ago.beginning_of_month) },
+        size_fn: -> { rand(1..3) },
+        build_fn: -> (args) {
+          now = args[:time_fn].call
+
+          build_blood_sugar_payload(FactoryBot.build(
+            :blood_sugar,
+            :with_hba1c,
+            patient: args[:patient],
+            user: user,
+            device_created_at: now,
+            device_updated_at: now,
+            facility: user.facility
+          ))
+        },
+        patient_sample_size: 0.05,
+        request_key: :blood_sugars,
+        api_version: 'v4'
+      },
+
+      scheduled_appointment: {
+        size_fn: -> { 1 },
+        build_fn: -> (args) {
+          return if args[:patient].latest_scheduled_appointment.present?
+
+          build_appointment_payload(FactoryBot.build(
+            :appointment,
+            patient: args[:patient],
+            user: user,
+            creation_facility: user.facility,
+            facility: user.facility
+          ))
+        },
+        patient_sample_size: 0.50,
+        request_key: :appointments,
+        api_version: 'v3'
+      },
+
+      overdue_appointment: {
+        size_fn: -> { rand(1..2) },
+        build_fn: -> (args) {
+          return if args[:patient].latest_scheduled_appointment.present?
+
+          build_appointment_payload(FactoryBot.build(
+            :appointment,
+            :overdue,
+            patient: args[:patient],
+            user: user,
+            creation_facility: user.facility,
+            facility: user.facility
+          ))
+        },
+        patient_sample_size: 0.50,
+        request_key: :appointments,
+        api_version: 'v3'
+      },
+
+      completed_phone_call: {
+        time_fn: -> { Faker::Time.between(from: 9.months.ago, to: Date.today) },
+        size_fn: -> { rand(1..1) },
+        build_fn: -> (args) {
+
+          FactoryBot.create(
+            :call_log,
+            result: 'completed',
+            caller_phone_number: user.phone_number,
+            callee_phone_number: args[:patient].latest_phone_number,
+            end_time: args[:time_fn].call
+          )
+        },
+        patient_sample_size: 0.20
+      }
+    }
+  end
+
+  def perform(user_id)
+    return if ENV['SIMPLE_SERVER_ENV'] == 'production'
```

Comment:
We should raise a custom error here, this will fail silently which would be bad.

Something like


```
class InvalidSeedOperation < RuntimeError; end

def perform(user_id)
  if Rails.env.production?
    raise InvalidSeedOperation, "Cannot generate seed data in Rails.env!"
  end
...
```

This should generate a flurry of Sentry errors, which would hopefully hit someone's pager if we have alerting on exception spikes. =). Or would at least make it very clear what the error spike was.

Reply:
[Addressed here.](https://github.com/simpledotorg/simple-server/pull/786/commits/8c30c3878505e0a30003dfc87ab64c870b2bc448)